desktop/core/src/desktop/log/api.py

```diff
@@ -0,0 +1,214 @@
+#!/usr/bin/env python
+# Licensed to Cloudera, Inc. under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  Cloudera, Inc. licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import time
+import socket
+import logging
+import zipfile
+import tempfile
+from wsgiref.util import FileWrapper
+
+from django.http import HttpResponse
+
+from desktop.auth.backend import is_admin
+from desktop.lib.django_util import JsonResponse
+from desktop.lib.i18n import smart_str
+from desktop.log import DEFAULT_LOG_DIR
+
+LOG = logging.getLogger()
+
+
+# TODO: Improve error response further with better context -- Error UX Phase 2
+def api_error_handler(view_fn):
+  """
+  Decorator to handle exceptions and return a JSON response with an error message.
+  """
+
+  def decorator(*args, **kwargs):
+    try:
+      return view_fn(*args, **kwargs)
+    except Exception as e:
+      LOG.exception(f'Error running {view_fn.__name__}: {str(e)}')
+      return JsonResponse({'error': str(e)}, status=500)
+
+  return decorator
+
+
+@api_error_handler
+def get_hue_logs(request):
+  """
+  Retrieves the last X characters of log messages from the log file.
+
+  Args:
+    request: The HTTP request object.
+
+  Returns:
+    A JSON response containing the current hostname and log messages.
+
+  Raises:
+    403: If user accessing the endpoint in not a Hue admin.
+    404: If the log directory or file does not exist.
+    500: If an internal server error occurs.
+
+  Notes:
+    This endpoint retrieves the last X characters of log messages from the log file.
+    The log file is read up to the buffer size, and if it's smaller than the buffer size,
+    the previous log file is read to complete the buffer.
+  """
+  if not is_admin(request.user):
+    return HttpResponse("You must be a Hue admin to access this endpoint.", status=403)
+
+  # Buffer size for reading log files
+  LOG_BUFFER_SIZE = 32 * 1024
+
+  log_directory = os.getenv("DESKTOP_LOG_DIR", DEFAULT_LOG_DIR)
+  if not log_directory:
+    return HttpResponse('The log directory is not set or does not exist.', status=404)
+
+  log_file = os.path.join(log_directory, 'rungunicornserver.log')
+  if not os.path.exists(log_file):
+    return HttpResponse('The log file does not exist.', status=404)
```

Comment:
its better to keep all the responses as JSON. i am not sure if this will return json response or just text.